src/test/java/org/opensearch/knn/training/VectorReaderTests.java

```diff
@@ -364,51 +363,69 @@ public void testRead_valid_NestedField() throws InterruptedException, ExecutionE
         VectorReader vectorReader = new VectorReader(client());
 
         // Read all vectors and confirm they match vectors
-        TestVectorConsumer testVectorConsumer = new TestVectorConsumer();
         final CountDownLatch inProgressLatch = new CountDownLatch(1);
+        TestFloatTrainingDataConsumer trainingDataConsumer = new TestFloatTrainingDataConsumer(createMockTrainingDataAllocation());
         vectorReader.read(
             clusterService,
             DEFAULT_INDEX_NAME,
             DEFAULT_NESTED_FIELD_PATH,
             DEFAULT_MAX_VECTOR_COUNT,
             DEFAULT_SEARCH_SIZE,
-            testVectorConsumer,
+            trainingDataConsumer,
             createOnSearchResponseCountDownListener(inProgressLatch)
         );
 
         assertLatchDecremented(inProgressLatch);
 
-        List<Float[]> consumedVectors = testVectorConsumer.getVectorsConsumed();
+        List<Float[]> consumedVectors = trainingDataConsumer.getTotalAddedVectors();
         assertEquals(DEFAULT_NUM_VECTORS, consumedVectors.size());
 
         List<Float> flatVectors = vectors.stream().flatMap(Arrays::stream).collect(Collectors.toList());
         List<Float> flatConsumedVectors = consumedVectors.stream().flatMap(Arrays::stream).collect(Collectors.toList());
         assertEquals(new HashSet<>(flatVectors), new HashSet<>(flatConsumedVectors));
     }
 
-    private static class TestVectorConsumer implements Consumer<List<Float[]>> {
+    private void assertLatchDecremented(CountDownLatch countDownLatch) throws InterruptedException {
+        assertTrue(countDownLatch.await(DEFAULT_LATCH_TIMEOUT, TimeUnit.SECONDS));
+    }
+
+    private ActionListener<SearchResponse> createOnSearchResponseCountDownListener(CountDownLatch countDownLatch) {
+        return ActionListener.wrap(response -> countDownLatch.countDown(), Throwable::printStackTrace);
+    }
+
+    private NativeMemoryAllocation.TrainingDataAllocation createMockTrainingDataAllocation() {
+        return new NativeMemoryAllocation.TrainingDataAllocation(null, 0, 0, VectorDataType.FLOAT);
+    }
 
-        List<Float[]> vectorsConsumed;
+    // create test float training data consumer class extending FloatTrainingDataConsumer
+    private static class TestFloatTrainingDataConsumer extends FloatTrainingDataConsumer {
```

Comment:
Why do we need this class? I think we can just use FloatTrainingDataConsumer?

Reply:
In the TestFloatTrainingDataConsumer `processTrainingVectors` method, it additionally count the total vectors used for test only.

Reply:
As I mentioned in another comment, I think we don't need to validate total vectors count. All we need is to verifying that JNIService is called with expected method and parameters.